flexboxAPIspecV2.md

```diff
@@ -0,0 +1,783 @@
+
+The order property specifies the order of a flexible item relative to the rest of the flexible items inside the same container.
+
+Note: If the element is not a flexible item, the order property has no effect.
+
+
+```xml
+        <controls:LayoutPanel>
+            <controls:LayoutPanel.Layout>
+                <controls:FlexboxLayout/>
+            </controls:LayoutPanel.Layout>
+            <Grid controls:FlexboxLayout.Order="5">
+                <TextBlock>5</TextBlock>
+            </Grid>
+            <Grid controls:FlexboxLayout.Order="-1">
+                <TextBlock>-1</TextBlock>
+            </Grid>
+            <Grid controls:FlexboxLayout.Order="2">
+                <TextBlock>2</TextBlock>
+            </Grid>
+            <Grid controls:FlexboxLayout.Order="1">
+                <TextBlock>1</TextBlock>
+            </Grid>
+        </controls:LayoutPanel>
+```
+
+### Output of the above code (styling omitted)
+![image](./FlexOrder.PNG)
+
+
+##  Flex Order Accepted Values
+
+
+
+| Value | Description |
+|-|-|
+| Number (Default Value = 0) | Specifies the order for the flexible item |
+
+# API Details
+
+
+```c# (but really MIDL3)
+
+unsealed runtimeclass FlexboxLayout : NonVirtualizingLayout
+    
+    {
+        static Windows.UI.Xaml.DependencyProperty OrderProperty{ get; };
+        static void SetOrder(Windows.UI.Xaml.UIElement element, Int32 value);
+        static Int32 GetOrder(Windows.UI.Xaml.UIElement element);
+    }
+```
+
+
+
+## Flex Grow
+
+The flex-grow property specifies how much the item will grow relative to the rest of the flexible items inside the same container.
+
+Note: If the element is not a flexible item, the flex-grow property has no effect.
+
+
+```xml
+ <controls:LayoutPanel MinWidth="500">
+            <controls:LayoutPanel.Layout>
+                <controls:FlexboxLayout/>
+            </controls:LayoutPanel.Layout>
+            <Grid controls:FlexboxLayout.Grow="0" controls:FlexboxLayout.Basis="100" >
+                <TextBlock>0</TextBlock>
+            </Grid>
+            <Grid controls:FlexboxLayout.Grow="1" controls:FlexboxLayout.Basis="100">
+                <TextBlock>1</TextBlock>
+            </Grid>
+            <Grid controls:FlexboxLayout.Grow="0" controls:FlexboxLayout.Basis="100">
+                <TextBlock>0</TextBlock>
+            </Grid>
+        </controls:LayoutPanel>
+```
+
+### Output of the above code (styling omitted)
+![image](./FlexGrow.PNG)
+
+
+##  Flex Grow Accepted Values
+
+
+
+| Value | Description |
+|-|-|
+| Number (Default Value = 0) | Specifies how much the item will grow relative to the rest of the flexible items. |
+
+# API Details
+
+
+```c# (but really MIDL3)
+
+unsealed runtimeclass FlexboxLayout : NonVirtualizingLayout
+    
+    {
+        static Windows.UI.Xaml.DependencyProperty GrowProperty{ get; };
+        static void SetGrow(Windows.UI.Xaml.UIElement element, Double value);
+        static Double GetGrow(Windows.UI.Xaml.UIElement element);
```

Comment:
we had a discussion about float/Single vs. double/Double
In general XAML properties of "number" types are Double, so maybe it would make sense to keep all these IDL properties as Double, and then update the code to not use float and use double instead.

If that doesn't work for some reason, then using Single would be how you represent C++ "float"

Reply:
Yes, default for numbers in Windows API is int or double

Reply:
Yes I did try changing everything to Single in the IDL file, but it ended up throwing errors and I left as is. Within the code changing everything to double is definitely an option